src/hyperdeck/hyperdeck-core.js

```diff
@@ -12,10 +12,34 @@ var logger = Logger.get('hyperdeck.HyperdeckCore');
  * Allows you to make requests to the hyperdeck and get its parsed responses.
  * This chains the requests so only one is sent at a time.
  * You can also listen for asynchronous events sent from the hyperdeck.
- * @param ip, The IP address of the hyperdeck.
+ * @param config hyperdeck configuration
+ *               config = 'ip address string[string]'
+ *               or
+ *               config = {
+ *                 ip: 'ip address string[string]',
+ *                 [ pingInterval: ping interval in miliseconds[int][default = 10000] ],
+ *                 [ timeout: inactivity timeout in miliseconds[int][default = 3000] ]
+ *               }
  **/
-function HyperdeckCore(ip) {
+function HyperdeckCore(config) {
+  if (config === null) {
```

Comment:
I think this should be

typeof config !== 'string' && typeof config !== 'object'

Reply:
see below